substrate/frame/asset-conversion/src/tests.rs

```diff
@@ -1685,3 +2024,429 @@ fn cannot_block_pool_creation() {
 		));
 	});
 }
+
+#[test]
+fn swap_transactional() {
+	new_test_ext().execute_with(|| {
+		let user = 1;
+		let user2 = 2;
+		let token_1 = NativeOrAssetId::Native;
+		let token_2 = NativeOrAssetId::Asset(2);
+		let token_3 = NativeOrAssetId::Asset(3);
+
+		let asset_ed = 150;
+		create_tokens_with_ed(user, vec![token_2, token_3], asset_ed);
+		assert_ok!(AssetConversion::create_pool(
+			RuntimeOrigin::signed(user),
+			Box::new(token_1),
+			Box::new(token_2)
+		));
+		assert_ok!(AssetConversion::create_pool(
+			RuntimeOrigin::signed(user),
+			Box::new(token_1),
+			Box::new(token_3)
+		));
+
+		let ed = get_ed();
+		assert_ok!(Balances::force_set_balance(RuntimeOrigin::root(), user, 20000 + ed));
+		assert_ok!(Assets::mint(RuntimeOrigin::signed(user), 2, user, 1000));
+		assert_ok!(Assets::mint(RuntimeOrigin::signed(user), 3, user, 1000));
+
+		assert_ok!(Balances::force_set_balance(RuntimeOrigin::root(), user2, 20000 + ed));
+		assert_ok!(Assets::mint(RuntimeOrigin::signed(user), 2, user2, 1000));
+		assert_ok!(Assets::mint(RuntimeOrigin::signed(user), 3, user2, 1000));
+
+		let liquidity1 = 10000;
+		let liquidity2 = 200;
+
+		assert_ok!(AssetConversion::add_liquidity(
+			RuntimeOrigin::signed(user),
+			Box::new(token_1),
+			Box::new(token_2),
+			liquidity1,
+			liquidity2,
+			1,
+			1,
+			user,
+		));
+
+		assert_ok!(AssetConversion::add_liquidity(
+			RuntimeOrigin::signed(user),
+			Box::new(token_1),
+			Box::new(token_3),
+			liquidity1,
+			liquidity2,
+			1,
+			1,
+			user,
+		));
+
+		let pool_1 = AssetConversion::get_pool_account(&(token_1, token_2));
+		let pool_2 = AssetConversion::get_pool_account(&(token_1, token_3));
+
+		assert_eq!(Balances::balance(&pool_1), liquidity1);
+		assert_eq!(Assets::balance(2, &pool_1), liquidity2);
+		assert_eq!(Balances::balance(&pool_2), liquidity1);
+		assert_eq!(Assets::balance(3, &pool_2), liquidity2);
+
+		// the amount that would cause a transfer from the last pool in the path to fail
+		let expected_out = liquidity2 - asset_ed + 1;
+		let amount_in = AssetConversion::balance_path_from_amount_out(
+			expected_out,
+			vec![token_2, token_1, token_3].try_into().unwrap(),
```

Comment:
just curious, why do you need to use `.try_into().unwrap()` given the `balance_path_from_amount_out()` accepts the unbounded vec now?

Reply:
no need, left here because it used to be a bounded vec, I will remove it